### PR TITLE
add folding apps

### PR DESCRIPTION
**Pull request by NiklasTR**
Major changes:

- [ ] added colabfold container
- [ ] added an element in the app.config that specifies wether a tool is using individual files, an index or a directory for inputs. Based on that parameter, the creation for index files is skipped when a model uses directories as inputs (batch processing built in). 
- [ ] used consistent quotation in the instruction template

### CODE REVIEW

**Comment by thetechnocrat-dev** [comment]
@NiklasTR can you update this branch with main and put a Loom or directions for testing?

**Comment by NiklasTR** [comment]
The branch is now up to date. After making sure the IFPS configuration is correct, run

```
bacalhau docker run --selector owner=labdao --gpu 1  --network full -i QmV18Lsn5QyFuZ1z1K24rt3eC7K5o1JsijNR3LPEgV27Bm public.ecr.aws/p7l9w5o7/colabfold:latest -- /bin/bash -c 'colabfold_batch --max-msa 32:64 --templates --num-recycle 1 /inputs /outputs/'
```

You can't run: 
```
./plex -app colabfold-mini -input-dir testdata/folding -gpu=true
```
Because the networking flag is not set right now

**Comment by NiklasTR** [comment]
Here are results:

```
bacalhau docker run --selector owner=labdao --gpu 1  --network full -i QmV18Lsn5QyFuZ1z1K24rt3eC7K5o1JsijNR3LPEgV27Bm public.ecr.aws/p7l9w5o7/colabfold:latest -- /bin/bash -c 'colabfold_batch --max-msa 32:64 --templates --num-recycle 1 /inputs /outputs/'
Job successfully submitted. Job ID: c85921ce-8d1f-4676-b54c-afbff085016a
Checking job status... (Enter Ctrl+C to exit at any time, your job will continue running):

           Communicating with the network  ................ done ✅  0.0s
              Finding node(s) for the job  ................ done ✅  0.0s
                          Running the job  ................ done ✅  10m00.2s
                          
                          
                          
bacalhau get c85921ce
```

File: cmd/plex/utils.go [hunk 1]
```diff
@@ -166,24 +167,32 @@ func createCombinations(indexMap map[string][]string, fieldA, fieldB string) []m
...
-		for _, input := range appConfig.Inputs {
-			for _, filetype := range input.Filetypes {
-				if strings.HasSuffix(filePath, filetype) {
-					indexMap[input.Field] = append(indexMap[input.Field], filePath)
+	// do not create index if the app config specifies the input method as "directory"
+	if appConfig.InputMethod == "directory" {
+		fmt.Println("Input method is directory, skipping index creation")
+		return "", []map[string]string{}
+	}
+	if appConfig.InputMethod == "directory" {
```
Comment (thetechnocrat-dev) [inline cmd/plex/utils.go:175]: I'm guessing this if statement logic is a typo?

should be `appConfig.Inputmethod != "directory"` or just put it in an else block attached to the first if statement.

**Review by thetechnocrat-dev — CHANGES_REQUESTED** [review]
It did run 🥳 . One small change requested.

Screen Shot 2023-03-03 at 11 35 44 AM

File: cmd/plex/utils.go (hunk 1 above)
Comment (NiklasTR) [reply cmd/plex/utils.go:175]: thank you - yes, I was lazy when patching this late in the day

**Review by NiklasTR — COMMENTED** [review]
reviewed

**Review by thetechnocrat-dev — APPROVED** [review]

**Pull request merged**